contracts/modules/capital/CowSwapOperator.sol

```diff
@@ -0,0 +1,240 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.9;
+
+import "../../external/cow/GPv2Order.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4/utils/math/Math.sol";
+import "../../interfaces/INXMMaster.sol";
+import "../../interfaces/IPool.sol";
+import "../../interfaces/IWeth.sol";
+import "../../interfaces/ICowSettlement.sol";
+import "../../interfaces/IPriceFeedOracle.sol";
+
+contract CowSwapOperator {
+  // Storage
+  ICowSettlement public immutable cowSettlement;
+  address public immutable cowVaultRelayer;
+  INXMMaster public immutable master;
+  address public immutable swapController;
+  IWeth public immutable weth;
+  IPriceFeedOracle public immutable priceFeedOracle;
+  bytes public currentOrderUID;
+  bytes32 public immutable domainSeparator;
+
+  // Constants
+  address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+  uint16 constant MAX_SLIPPAGE_DENOMINATOR = 10000;
+  uint256 constant MIN_VALID_TO_PERIOD = 600; // 10 minutes
+  uint256 constant MAX_VALID_TO_PERIOD = 3600; // 60 minutes
+  uint256 constant MIN_SELL_AMT_TO_FEE_RATIO = 100; // Sell amount at least 100x fee amount
+
+  event OrderPlaced(GPv2Order.Data order);
+  event OrderClosed(GPv2Order.Data order, uint256 filledAmount);
+
+  modifier onlyController() {
+    require(msg.sender == swapController, "SwapOp: only controller can execute");
+    _;
+  }
+
+  constructor(
+    address _cowSettlement,
+    address _swapController,
+    address _master,
+    address _weth,
+    address _priceFeedOracle
+  ) {
+    cowSettlement = ICowSettlement(_cowSettlement);
+    cowVaultRelayer = cowSettlement.vaultRelayer();
+    master = INXMMaster(_master);
+    swapController = _swapController;
+    weth = IWeth(_weth);
+    priceFeedOracle = IPriceFeedOracle(_priceFeedOracle);
```

Comment:
Price feed oracle is not immutable in the Pool contract and may be updated. Instead of having it immutable or storage - read it from the pool contract each time a new order is placed.